Admin/BreadcrumbsBuilderInterface.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+use Knp\Menu\ItemInterface;
+
+/**
+ * Builds a breacrumbs. There is a dependency on the AdminInterface because
+ * this object holds useful object to deal with this task, but there is
+ * probably a better design.
+ *
+ * @author Grégoire Paris <postmaster@greg0ire.fr>
+ */
+interface BreadcrumbsBuilderInterface
+{
+    /**
+     * Get breadcrumbs for $action.
+     *
+     * @param AdminInterface $admin
+     * @param string         $action the name of the action we want to get a
+     *                               breadcrumbs for
+     *
+     * @return mixed ItemInterface the breadcrumbs
```

Comment:
Wow good catch, I change that to array|Traversable (don't care what it 
is as long as it is usable in a foreach).

Le 23/05/2016 15:11, Oskar Stark a écrit :
>
> In Admin/BreadcrumbsBuilderInterface.php 
> <https://github.com/sonata-project/SonataAdminBundle/pull/3791#discussion_r64219053>:
>
> > + * Builds a breacrumbs. There is a dependency on the AdminInterface because
> > + * this object holds useful object to deal with this task, but there is
> > + * probably a better design.
> > + *
> > + * @author Grégoire Paris <postmaster@greg0ire.fr>
> > + */
> > +interface BreadcrumbsBuilderInterface
> > +{
> > +    /**
> > +     * Get breadcrumbs for $action.
> > +     *
> > +     * @param AdminInterface $admin
> > +     * @param string         $action the name of the action we want to get a
> > +     *                               breadcrumbs for
> > +     *
> > +     * @return mixed ItemInterface the breadcrumbs
>
> mixed of what?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/sonata-project/SonataAdminBundle/pull/3791/files/25b1b897508798ba6456c2a8ff65a340048c1213#r64219053>
>